pkg/engine/imageVerify.go

```diff
@@ -25,6 +25,61 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+func hasMatchedImages(images map[string]map[string]apiutils.ImageInfo, rule *kyvernov1.Rule) bool {
```

Comment:
Can we return the set of matching images here, so we do not have to perform the same checks, and run the image extractors, again later in the flow?

If we can find the matching set of images up front, we can then pass them to the code that validates each image.

Reply:
fixed